docs/platform/porting/system.md

```diff
@@ -2540,3 +2540,356 @@ Example:
+
+-  **foreground_web_app**  
+     Web applications running in foreground
+
+-  **background_web_app**  
+     Web applications running in background
+
+-  **foreground_csharp_app**  
+     .NET applications running in foreground
+
+-  **background_csharp_app**  
+     .NET applications running in background
+
+-  **foreground_ui_widget**  
+     Widget applications running in foreground
+
+-  **background_ui_widget**  
+     Widget applications running in background
+
+-  **native_service**  
+     Service applications
+
+-  **other_service**  
+     Other system processes
+
+`global` is the ruleset used as a fallback when neither program-specific nor the relevant class-wide configuration exists. In addition to the rules described below, this ruleset should specify the `sampling_rate` value which describes how often (in Hz) to sample program parameters (actual reads from the kernel).
+
+Each ruleset specifies a set of parameters to monitor for given program. These are:
+
+-  **monitor**  
+     Defines whether to monitor a process.
+
+     Default value: `1`
+
+-  **print_current**  
+     Defines whether to print current process' resource usage in realtime.
+
+     Default value: `0`
+
+-  **apply_to_children**  
+     Defines whether process' ruleset should be applied to all of its children.
+
+     Default value: `0`
+
+-  **kill**  
+     Defines whether to kill the process after detecting abnormality.
+
+     Default value: `0`
+
+-  **report**  
+     Defines whether to create bugreport after detecting abnormality.
+
+     Default value: `0`
+
+-  **sampling_rate**  
+     Specifies the sampling rate (Hz). This should not be more often than the value specified in the global section. This limits how much processing is done on data read from kernel, but not the actual reads from kernel (for example if the global section specifies sampling rate 2 and the local section specifies a sampling rate of 1, data will be read from the kernel twice per second but only processed once per second - the second batch of data will be discarded).
+
+     Default value: `1`
+
+-  **cpu_limit_peak**  
+     Specifies the maximum allowed CPU usage in any given sample, where a value of 1.0 corresponds to the equivalent of full use of one core. Note that this is a total and cannot disambiguate between individual cores, and does not know how many cores there are. This means that the same value can represent widely different situations so a lot of care needs to be taken when picking a value. Consider a value of 2.0 and how it could be reached:
+     -  a 2-core system where both cores are at 100%. This is extremely heavy usage of the CPU.
+     -  an 8-core system where two cores are at 100% and the remaining six are at 0%. This is a bit worrisome but much less of a danger to the system than the above.
+     - an 8-core system where all eight cores are at 25%. This is quite a well-behaved application compared to the two.
+
```

Comment:
This paragraph is too hard to understand for readers. Please paraphrase.
Please use subjects instead of pronouns.